modules/js-and-css/speculation-rules/settings.php

```diff
@@ -0,0 +1,218 @@
+		'prefetch'  => _x( 'Prefetch', 'setting label', 'performance-lab' ),
+		'prerender' => _x( 'Prerender', 'setting label', 'performance-lab' ),
+	);
+}
+
+/**
+ * Returns the available options for the Speculation Rules eagerness and their labels.
+ *
+ * @since n.e.x.t
+ *
+ * @return array Associative array of `$eagerness => $label` pairs.
+ */
+function plsr_get_eagerness_labels() {
+	return array(
+		'conservative' => _x( 'Conservative (on click)', 'setting label', 'performance-lab' ),
+		'moderate'     => _x( 'Moderate (on hover)', 'setting label', 'performance-lab' ),
+		'eager'        => _x( 'Eager (on slightest suggestion)', 'setting label', 'performance-lab' ),
+	);
+}
+
+/**
+ * Returns the default setting value for Speculation Rules configuration.
+ *
+ * @since n.e.x.t
+ *
+ * @return array Default value, an associative array with 'mode' and 'eagerness' keys.
+ */
+function plsr_get_setting_default() {
+	return array(
+		'mode'      => 'prerender',
+		'eagerness' => 'moderate',
+	);
+}
+
+/**
+ * Sanitizes the setting for Speculation Rules configuration.
+ *
+ * @since n.e.x.t
+ *
+ * @param mixed $input Setting to sanitize.
+ * @return array Sanitized setting, an associative array with 'mode' and 'eagerness' keys.
+ */
+function plsr_sanitize_setting( $input ) {
+	$default_value = plsr_get_setting_default();
+
+	if ( ! is_array( $input ) ) {
+		return $default_value;
+	}
+
+	$mode_labels      = plsr_get_mode_labels();
+	$eagerness_labels = plsr_get_eagerness_labels();
+
+	// Ensure only valid keys are present.
+	$value = array_intersect_key( $input, $default_value );
+
+	// Set any missing or invalid values to their defaults.
+	if ( ! isset( $value['mode'] ) || ! isset( $mode_labels[ $value['mode'] ] ) ) {
+		$value['mode'] = $default_value['mode'];
+	}
+	if ( ! isset( $value['eagerness'] ) || ! isset( $eagerness_labels[ $value['eagerness'] ] ) ) {
+		$value['eagerness'] = $default_value['eagerness'];
+	}
+
+	return $value;
+}
+
+/**
+ * Registers setting to control Speculation Rules configuration.
+ *
+ * @since n.e.x.t
+ * @access private
+ */
+function plsr_register_setting() {
+	register_setting(
+		'reading',
+		'plsr_speculation_rules',
+		array(
+			'type'              => 'object',
+			'description'       => __( 'Configuration for the Speculation Rules API.', 'performance-lab' ),
+			'sanitize_callback' => 'plsr_sanitize_setting',
+			'default'           => plsr_get_setting_default(),
+			'show_in_rest'      => array(
+				'schema' => array(
+					'properties' => array(
+						'mode'      => array(
+							'description' => __( 'Whether to prefetch or prerender URLs.', 'performance-lab' ),
+							'type'        => 'string',
+							'enum'        => array_keys( plsr_get_mode_labels() ),
+						),
+						'eagerness' => array(
+							'description' => __( 'Whether to trigger on click (conservative), on hover (moderate), or on even the slight suggestion (eager) that the user may navigate to the URL.', 'performance-lab' ),
```

Comment:
if any change is made above it should be reflected here

Reply:
Is there a good way for me to stay in the loop on this? If not, it would be great if you could ping me should there be related discussions.

Reply:
There's an internal mailing list where Chrome stuff spanning this area is discussed, and sometimes that includes status updates (though it also has other discussion that may be noise to you). I'm in the process of refreshing an evergreen status doc, too. I'll try to remember to send it to you when done.